hidi/matrix.py

```diff
@@ -226,6 +227,74 @@ def transform(self, M,  **kwargs):
         return self.keras_model, kwargs
 
 
+class KerasKfoldTransform(Transform):
+    """
+    Generalized transform for Keras algorithm with k fold cross validation
+    evaluation
+    """
+    def __init__(self, keras_model, validation_matrix, tensorboard_log_dir,
+                 kfold_n_splits=10, kfold_seed=42, kfold_shuffle=True,
+                 classification=False, **keras_kwargs):
+        self.keras_model = keras_model
+        self.keras_kwargs = keras_kwargs
+        self.validation_matrix = validation_matrix
+
+        self.kfold_n_splits = kfold_n_splits
+        self.kfold_seed = kfold_seed
+        self.kfold_shuffle = kfold_shuffle
+
+        self.log_dir = tensorboard_log_dir
+
+        self.classification = classification
+
+        if 'item_id' in validation_matrix.columns:
+            self.validation_matrix.set_index('item_id', inplace=True)
+
+    def transform(self, M,  **kwargs):
+        """
+        Takes a Takes a dataframe that has :code:`item_id` index, other
+        'features' columns for prediction, and applies a Keras sequential
+        model to it.
+
```

Comment:
Add parameter and return value documentation similar to [this code](https://github.com/VEVO/hidi/blob/e266f945f0868fffeeb157da0abf232a16444b9d/hidi/matrix.py#L132-L142).